ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/service/impl/ToolAdminServiceFacadeImpl.java

```diff
@@ -0,0 +1,111 @@
+    @Inject private DiscEnvApiService deService;
+
+    @Override
+    public void getTools(String searchTerm, AsyncCallback<List<Tool>> callback) {
+        String address = TOOLS + "?search=" + URL.encodeQueryString(searchTerm);
+
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, address);
+        deService.getServiceData(wrapper, new ToolListCallbackConverter(callback, factory));
+    }
+
+    @Override
+    public void getToolDetails(String toolId, AsyncCallback<Tool> callback) {
+        String address = TOOLS + "/" + toolId;
+
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, address);
+        deService.getServiceData(wrapper, new ToolCallbackConverter(callback, factory));
+    }
+
+    @Override
+    public void addTool(ToolList toolList, AsyncCallback<Void> callback) {
+        String address = TOOLS_ADMIN;
+
+        final Splittable encode = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(toolList));
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(POST, address, encode.getPayload());
+        deService.getServiceData(wrapper, new StringToVoidCallbackConverter(callback));
+    }
+
+    @Override
+    public void updateTool(Tool tool, boolean overwrite, AsyncCallback<Void> callback) {
+
+        String toolId = tool.getId();
+        String address;
+        if (overwrite) {
+            address = TOOLS_ADMIN + "/" + toolId + "?overwrite-public=true";
+        } else {
+            address = TOOLS_ADMIN + "/" + toolId;
+        }
+
+        final Splittable encode = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(tool));
+        String payload = removeEmptyValues(encode);
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(PATCH, address, payload);
+        deService.getServiceData(wrapper, new StringToVoidCallbackConverter(callback));
+    }
+
+    @Override
+    public void deleteTool(String toolId, AsyncCallback<Void> callback) {
+        String address = TOOLS_ADMIN + "/" + toolId;
+
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(DELETE, address);
+        deService.getServiceData(wrapper, new StringToVoidCallbackConverter(callback));
+    }
+
+    private String removeEmptyValues(Splittable payload) {
+        removeEmptyValuesHelper(payload);
+        return payload.getPayload();
+    }
+
+    private void removeEmptyValuesHelper(Splittable obj) {
+        List<String> propertyKeys = obj.getPropertyKeys();
+
+        for (String key : propertyKeys) {
+            Splittable value = obj.get(key);
+            String temp = value.getPayload().replaceAll("[\\[\\]\\{\\}]", "");
+            if ((temp.equals("") || temp.equals("\"\"")) && !key.matches("(input|output)_files")) {
+                StringQuoter.nullValue().assign(obj, key);
+            } else {
+                if (value.isIndexed() || value.isKeyed()) {
+                    removeEmptyValuesHelper(value);
+                }
+            }
+        }
+    }
+}
```

Comment:
This is generally the way our `PATCH` endpoints work. If the value is not submitted in the `PATCH` request, then it is not updated. Most of the `container` subfields are different, where if the `container` subfield is not submitted as part of the `container`, it gets deleted.

From this `PATCH` endpoint's docs:
> Note: If the `container` object is omitted in the request, then existing container settings will not be modified, but if the `container` object is present in the request, then all container settings must be included in it. Any existing settings not included in the request's `container` object will be removed.

In all this discussion about not submitting blank strings, this point should have been clarified. Since the `container` object works this way, those fields should not be submitted with blank strings, but for other top-level fields like `attribution` or `version`, there's not another way to remove the value with this endpoint, so those fields can be submitted with empty strings in order to "delete" those values.